tooling/lsp/src/notifications/mod.rs

```diff
@@ -127,6 +141,10 @@ pub(super) fn on_did_save_text_document(
                 Err(errors_and_warnings) => errors_and_warnings,
             };
 
+            let package_root_dir: String = package.root_dir.as_os_str().to_string_lossy().into();
+
+            state.cached_definitions.insert(package_root_dir, context.def_interner.clone());
```

Comment:
On second thoughts, we could avoid this clone if we shift the caching to the end of this function. `Context` isn't mutated after `check_crate` so this shouldn't cause any issues.

Reply:
Just to be clear, this isn't a requirement. Documenting on `Trait` would resolve this too.

Reply:
improved here
f821f06: chore: drop clone on node_interner
TY